simplex/bls.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/simplex"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+)
+
+var (
+	errSignatureVerificationFailed = errors.New("signature verification failed")
+	errSignerNotFound              = errors.New("signer not found in the membership set")
+	simplexLabel                   = []byte("simplex")
+)
+
+var _ simplex.Signer = (*BLSSigner)(nil)
+
+type SignFunc func(msg []byte) (*bls.Signature, error)
+
+// BLSSigner signs messages encoded with the provided ChainID and SubnetID.
+// using the SignBLS function.
+type BLSSigner struct {
+	chainID  ids.ID
+	subnetID ids.ID
+	// signBLS is passed in because we support both software and hardware BLS signing.
+	signBLS SignFunc
+}
+
+type BLSVerifier struct {
+	nodeID2PK map[ids.NodeID]bls.PublicKey
```

Comment:
Throughout the repo, we normally pass `bls.PublicKey`s as pointers. Would it make sense to stay consistent with that?